packages/api/src/api/queue.ts

```diff
@@ -27,3 +29,39 @@ queue.get('/:type', async (req, res) => {
     res.status(500).send(errorMsg);
   }
 });
+
+queue.post('/join', async (req, res) => {
+  const user = req.userEntity;
+  const { type } = req.body;
+
+  if (!(type in QueueType)) {
+    res.status(400).send('The queue type entered is invalid');
+    return;
+  }
+
+  try {
+    const currentQueueItems = await req.entityManager.find(QueueUser, {
+      user,
+      status: { $in: ['Pending', 'InProgress'] },
+    });
```

Comment:
Let's change this call to a count, that way call is faster
```suggestion
    const userAlreadyInQueue = await req.entityManager.count(QueueUser, {
      user,
      status: { $in: ['Pending', 'InProgress'] },
    }).then((count) => !!count);
```